config.js

```diff
@@ -1,79 +1,112 @@
+    default: '',
+    envTemplate: 'DB_{database}_AUTH_MECHANISM'
   },
-  authDatabase: {
-    doc: "The database to authenticate against when supplying a username and password",
+  maxPoolSize: {
+    doc: 'The maximum number of connections in the connection pool',
+    format: Number,
+    default: 0,
+    envTemplate: 'DB_{database}_MAX_POOL'
+  },
+  password: {
+    doc: '',
     format: String,
-    default: "admin",
-    env: "DB_AUTH_SOURCE"
+    default: '',
+    envTemplate: 'DB_{database}_PASSWORD'
   },
-  database: {
-    doc: "",
+  readPreference: {
+    doc: 'Choose how MongoDB routes read operations to the members of a replica set - see https://docs.mongodb.com/manual/reference/read-preference/',
+    format: ['primary', 'primaryPreferred', 'secondary', 'secondaryPreferred', 'nearest'],
+    default: 'secondaryPreferred'
+  },
+  replicaSet: {
+    doc: '',
     format: String,
-    default: "test",
-    env: "DB_NAME"
+    default: ''
   },
   ssl: {
-    doc: "",
+    doc: '',
     format: Boolean,
     default: false
   },
-  replicaSet: {
-    doc: "",
+  username: {
+    doc: '',
     format: String,
-    default: ""
+    default: '',
+    envTemplate: 'DB_{database}_USERNAME'
+  }
+}
+
+const MAIN_SCHEMA = {
+  database: {
+    default: 'api',
+    doc: 'The name of the default database to be used',
+    env: 'DB_NAME',
+    format: String
   },
-  readPreference: {
-    doc: "Choose how MongoDB routes read operations to the members of a replica set - see https://docs.mongodb.com/manual/reference/read-preference/",
-    format: ['primary', 'primaryPreferred', 'secondary', 'secondaryPreferred', 'nearest'],
-    default: 'secondaryPreferred'
+  databases: {
+    default: {},
+    doc: 'Configuration block for each of the databases used throughout the application',
+    format: Object
   },
   enableCollectionDatabases: {
-    doc: "",
-    format: Boolean,
-    default: false
+    default: false,
+    doc: 'Whether to use a database specified in the collection endpoint',
+    format: Boolean
+  },
+  env: {
+    arg: 'node_env',
+    default: 'development',
+    doc: 'The applicaton environment.',
+    env: 'NODE_ENV',
+    format: ['production', 'development', 'test', 'qa']
   }
-})
+}
 
-// Load environment dependent configuration
-var env = conf.get('env')
-conf.loadFile('./config/mongodb.' + env + '.json')
+const mainConfig = convict(MAIN_SCHEMA)
+
+// Load environment dependent configuration.
+const environment = mainConfig.get('env')
+
+mainConfig.loadFile(`./config/mongodb.${environment}.json`)
+mainConfig.validate()
+
+// Validating databases.
+const databases = mainConfig.get('databases')
+
+Object.keys(databases).forEach(databaseName => {
+  const databaseConfig = convict(DATABASE_SCHEMA)
+
+  databaseConfig.load(databases[databaseName])
+  databaseConfig.validate()
+
+  const schema = databaseConfig.getSchema().properties
+
+  // Listening for database-specific environment variables.
+  // e.g. DB_testdb_USERNAME
```

Comment:
I wonder if we should enforce capitalisation here, so the env var might be `DB_{database.toUpperCase()}_USERNAME`

Reply:
Agree! Will change.